app/models/embedded_ansible_worker.rb

```diff
@@ -18,6 +19,7 @@ def start_runner
       end
     end
   end
+  alias_method :start_runner_via_spawn, :start_runner_via_fork
```

Comment:
Not an excuse to not include @carbonin 's input, but my two cents for how this should be done is leave `start_runner` as the method that is overwritten by this class, and slightly change up how it is implemented in the main `MiqWorker` class:

```diff
  def start_runner
+   ENV['MIQ_SPAWN_WORKERS'].to_s.downcase == "true" ? start_runner_via_spawn : start_runner_via_fork
+ end
+
+ def start_runner_via_fork
    self.class.before_fork
    pid = fork(:cow_friendly => true) do
      self.class.after_fork
 @@ -346,8 +346,25 @@ def start_runner
    pid
  end
  
+ def self.build_command_line(guid)
+   command_line = "#{Gem.ruby} #{runner_script} --heartbeat --guid=#{guid} #{self.name}"
+   ENV['APPLIANCE'] ? "nice #{nice_increment} #{command_line}" : command_line
+ end
+
+ def self.runner_script
+   script = ManageIQ.root.join("lib/workers/bin/run_single_worker.rb")
+   raise "script not found: #{script}" unless File.exist?(script)
+   script
+ end
+
+ def start_runner_via_spawn
+   self.pid = Kernel.spawn(self.class.build_command_line(guid), :out => "/dev/null", :err => [Rails.root.join("log", "evm.log"), "a"])
+   Process.detach(pid)
+   pid
+ end
+ 
  def start
   self.pid = start_runner
 ```

This way, any changes that are necessary to the normal workflow are done in the edgecase classes, and attempted to be accounted for in the base class.